### PR TITLE
adding interface for the f guild keys

### LINKED ISSUE #913: Interface for keeping track of the guild's keys
Create something similar to the tools interface, but for keeping track of keys instead.

### PR DESCRIPTION
**Pull request by Lullebullelukas**
An interface to create and manage keys that belong to the guild.
Closes #913 
:)))) xoxo

### CODE REVIEW

File: app/controllers/admin/keys_controller.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    redirect_to admin_keys_path, notice: alert_destroy(KeyUser)
+  end
+
+  def destroy
+    key = Key.find(params[:id])
+    if key.key_users.size == 0
+      key.destroy!
+      redirect_to admin_keys_path, notice: alert_destroy(Key)
+    else
+      redirect_to admin_keys_path,  notice: alert_danger('Kan inte ta bort nyckel,
```
[the remark this answers is not in the document]
Comment (Lullebullelukas) [reply app/controllers/admin/keys_controller.rb]: wat do

File: app/controllers/admin/keys_controller.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  def key_user_params
+    params.require(:key_user).permit(:user_id, :key_id)
+  end
+
+  def key_params
+    params.require(:key).permit(:name, :description, :total, :users)
+  end
+
+  def update
```
[the remark this answers is not in the document]
Comment (Lullebullelukas) [reply app/controllers/admin/keys_controller.rb]: wat do

File: app/controllers/admin/keys_controller.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    @key = Key.find(params[:id])
+    if params[:key][:total].to_i >= @key.key_users.size
+      if @key.update(key_params)
```
[the remark this answers is not in the document]
Comment (Lullebullelukas) [reply app/controllers/admin/keys_controller.rb]: wat do

File: app/controllers/admin/keys_controller.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        redirect_to admin_keys_path, notice: alert_update(Key)
+      end
+    else
+      redirect_to edit_admin_key_path, notice: alert_danger('Det finns mer utlåningar än önskat antal nycklar')
```
[the remark this answers is not in the document]
Comment (Lullebullelukas) [reply app/controllers/admin/keys_controller.rb]: wat do

File: app/controllers/admin/keys_controller.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    redirect_to admin_keys_path, notice: alert_destroy(KeyUser)
+  end
+
+  def destroy
+    key = Key.find(params[:id])
+    if key.key_users.empty?
+      key.destroy!
+      redirect_to admin_keys_path, notice: alert_destroy(Key)
+    else
+      redirect_to admin_keys_path,  notice: alert_danger('Kan inte ta bort nyckel, ta bort utlåningarna först')
```
[the remark this answers is not in the document]
Comment (davidkrantz) [reply app/controllers/admin/keys_controller.rb]: Fix spacing.

File: app/views/admin/keys/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<div class="col-md-8 col-md-offset-2 col-sm-12 reg-page key">
+    <div class="headline">
+      <h1><%= @key.name %></h1>
+    </div>
+    <div class="content">
+      <p> <span>Beskrivning:</span> <%= @key.description %> </p>
+      <p> <span>Totalt antal nycklar:</span> <%= @key.total %> </p>
+      <p> <span>Totalt antal lediga nycklar:</span> <%= @key.free_count %> </p>
```
[the remark this answers is not in the document]
Comment (Lullebullelukas) [reply app/views/admin/keys/show.html.erb]: this I liked very nice :)

**Pull request merged**